File: 242.有效的字母异位词.cpp
/*
 * @lc app=leetcode.cn id=242 lang=cpp
 *
 * [242] 有效的字母异位词
 */

// @lc code=start
class Solution {
public:
    bool isAnagram(string s, string t) {
        int sa[26] = {0};
        int len1 = s.length();
        int len2 = t.length();
        if (len1 != len2) return false;
        for (int i = 0; i < len1; i++) {
            sa[s[i] - 'a']++;
            sa[t[i] - 'a']--;
        }
        for (int i = 0; i < 26; i++) {
            if (sa[i] != 0) {
                return false;
            }
        }
        return true;
    }
};
// @lc code=end

File: 242.有效的字母异位词.go
/*
 * @lc app=leetcode.cn id=242 lang=golang
 *
 * [242] 有效的字母异位词
 */

// @lc code=start

func isAnagram(s string, t string) bool {
	m:= map[int32]int{}
	for _, c := range s {
		m[c]++
	}

	for _, c := range t {
		m[c]--
	}

	for _, v := range m {
		if v != 0 {
			return false
		}
	}
	return true
}
// @lc code=end

